src/viur/core/modules/user.py

```diff
@@ -619,10 +623,28 @@ def login(self, token, *args, **kwargs):
         return self._user_module.continueAuthenticationFlow(self, userSkel["key"])
 
 
-class TimeBasedOTP(UserAuthentication):
+class UserSecondFactorAuthentication(UserAuthentication, abc.ABC):
+    second_factor_login_template = "user_login_secondfactor"
+    """Template to enter the TOPT on login"""
+
+    NAME = ""
+    ACTION_NAME = ""
+
+    def __init__(self, moduleName, modulePath, _user_module):
+        super().__init__(moduleName, modulePath, _user_module)
+        self.action_url = f"{self.modulePath}/{self.ACTION_NAME}"
+        self.start_url = f"{self.modulePath}/start"
+
+    def can_handle(self, possible_user: db.Entity):
```

Comment:
```suggestion
    @abc.abstractmethod
    def can_handle(self, possible_user: db.Entity) -> bool:
```
Otherwise using `ABC` makes no sense...